lib/xnvme_buf.c

```diff
@@ -18,6 +18,8 @@ xnvme_buf_virt_alloc(size_t alignment, size_t nbytes)
 #ifdef WIN32
 	return _aligned_malloc(nbytes, alignment);
 #else
+	// nbytes has to be a multiple of alignment. Therefore, we round up to the nearest multiple.
```

Comment:
This would be great as commit in itself with a commit-message like:
```
buf: defensively meet alignment constraints in xnvme_buf_virt_alloc()

The virtual memory allocator uses aligned_alloc()/_aligned_mmalloc().
These require that nbytes is a positive multiple of alignment, this change ensures that.
```

And then move the re-computation of ``nbytes`` up in between 17-18, such that it also applies to Windows.

Reply:
Then for the meson-changes, something like:
```
build:meson: enable build of be:posix on Darwin/macOS
```